File: Minimum_Number_of_Operations_to_Make_Array_Empty.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        unordered_map<int, int> freq;

        for (int n : nums) {
            freq[n] = 1 + freq[n];
        }

        if (any_of(freq.begin(), freq.end(), [](const auto& kv) { return kv.second == 1; })) {
            return -1;
        }

        return accumulate(freq.begin(), freq.end(), 0, [](int sum, const auto& kv) { return sum + (kv.second + 2) / 3; });        
    }
};

File: Minimum_Number_of_Operations_to_Make_Array_Empty.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minOperations = function(nums) {
    const freq = new Map();

    for (const n of nums) {
        freq.set(n, 1 + (freq.get(n) || 0));
    }

    if ([...freq.values()].includes(1)) {
        return -1;
    }

    return [...freq.values()].reduce((sum, f) => sum + Math.floor((f + 2) / 3), 0);    
};
